sound/soc/sof/topology.c

```diff
@@ -2057,14 +2057,31 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
 	dev_dbg(sdev->dev, "tplg: config HDA%d fmt 0x%x\n",
 		config->dai_index, config->format);
 
-	/* send message to DSP */
-	ret = sof_ipc_tx_message(sdev->ipc,
-				 config->hdr.cmd, config, size, &reply,
-				 sizeof(reply));
+	if (link->dpcm_playback) {
+		snd_sof_dai_config(sdev, link, config, SNDRV_PCM_STREAM_PLAYBACK);
```

Comment:
comment related to the others, is it needed to have an sdev-level indirection when this is only called in the case of HDaudio and on a per-DAI basis? It feels like a private hda-specific case, not a global callback.

Reply:
yes, It is ugly for a issue in FW, but later I find another way to do it. I will refine it